app/src/lib/stores/helpers/branch-pruner.ts

```diff
@@ -66,11 +67,21 @@ export class BranchPruner {
     defaultBranch: Branch
   ): Promise<ReadonlyArray<string> | null> {
     const gitStore = this.gitStoreCache.get(repository)
-    return (
+    const mergedBranches =
       (await gitStore.performFailableOperation(() =>
         getMergedBranches(repository, defaultBranch.name)
       )) || null
-    )
+
+    // remove the current branch
+    if (mergedBranches !== null) {
+      const { branchesState } = this.repositoriesStateCache.get(repository)
+      const { tip } = branchesState
+      if (tip.kind === TipState.Valid) {
+        return mergedBranches.filter(x => x !== tip.branch.name)
```

Comment:
`tip.branch.name` currently returns a ref in the form of `dev` instead of what we need, `refs/heads/dev`. The `Branch` model may need to be updated to support this first.

Reply:
To avoid touching the model `nameWithoutRemote` should be safe enough to use. I also suggest appending this to the reserved branches list rather than adding it in here:

```diff
diff --git a/app/src/lib/stores/helpers/branch-pruner.ts b/app/src/lib/stores/helpers/branch-pruner.ts
index 2087aa71c..efab216b6 100644
--- a/app/src/lib/stores/helpers/branch-pruner.ts
+++ b/app/src/lib/stores/helpers/branch-pruner.ts
@@ -10,6 +10,7 @@ import {
 import { fatalError } from '../../fatal-error'
 import { RepositoryStateCache } from '../repository-state-cache'
 import * as moment from 'moment'
+import { TipState } from '../../../models/tip'
 
 /** Check if a repo needs to be pruned at least every 4 hours */
 const BackgroundPruneMinimumInterval = 1000 * 60 * 60 * 4
@@ -100,7 +101,7 @@ export class BranchPruner {
 
     // Get list of branches that have been merged
     const { branchesState } = this.repositoriesStateCache.get(this.repository)
-    const { defaultBranch } = branchesState
+    const { defaultBranch, tip } = branchesState
 
     if (defaultBranch === null) {
       return
@@ -130,9 +131,14 @@ export class BranchPruner {
       twoWeeksAgo
     )
 
+    const branchesToExclude = [...ReservedBranches]
+    if (tip.kind === TipState.Valid) {
+      branchesToExclude.push(`refs/heads/${tip.branch.nameWithoutRemote}`)
+    }
+
     // Create array of branches that can be pruned
     const candidateBranches = mergedBranches.filter(
-      b => !ReservedBranches.includes(b)
+      b => !branchesToExclude.includes(b)
     )
     const branchesReadyForPruning = new Array<Branch>()
     for (const branch of candidateBranches) {
```

Reply:
@niik I thought because `for-each-ref` was only looking at `refs/heads` it wouldn't do this:

https://github.com/desktop/desktop/blob/3f2084e1cb88a08abc9e5cf8123ef256a59096f8/app/src/lib/git/for-each-ref.ts#L40-L42

A quick test locally creating a `release-0.0.3` branch to clash shows that it's ensuring the shortname is unique:

```
~/src/desktop> git for-each-ref refs/heads --format="%(refname) %(refname:short)" | grep release-0.0.3
refs/heads/release-0.0.3 heads/release-0.0.3
```

👍 to getting `git branch --merged` to emit canonical refs and avoiding working with local branches, if feasible

Reply:
Fixed in d0cb2ef